### DISCUSSION TITLE
Anleitung für Home Assistant

### DISCUSSION THREAD

**Discussion by Marow** [syssi/esphome-atorch-dl24#47] [General]
Hi,

gibt es so eine Art "Anleitung" für Home Assistant? Also wie ich das dort einbinden könnte?

Habe bereits ESP Home dort installiert und tracke damit z.B. meine Gaszähler. Aber bin jetzt technisch nicht so weit, das ich wüsste, wie ich meinen DL24 dort mit reinbekomme. 

Kurzes Tutorial für Home Assistant mit dem Standard ESPHome wäre echt cool :)

**Comment by syssi** [comment]
Na klar! Ich schreibe dir gleich die wichtigsten Punkte auf. Du wirst dich wundern, weil du den gesamten Prozess bereits für deinen Gaszähler 1:1 geklickt haben müsstest.

**Reply by syssi** [reply]
Hier eine Kurzanleitung für die Einbindung eines Atorch Dummyloads (z.B. zur kontrollierten Entladung von LiFePo4-Zellen) in Home Assistant. Ich habe selbst keine Erfahrung mit dem eingebetteten ESPHome Dashboard in Home Assistant. Deshalb verzeih mir, wenn ich hier generisch bleibe. Du wirst diesen Teil aber sicher zuhauf im Netz finden.

1. https://raw.githubusercontent.com/syssi/esphome-atorch-dl24/main/esp32-dc-meter-example.yaml herunterladen und in einem Editor öffnen.
2. Diese Abschnitt entfernen

```
mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client
```

Da unsere Node nicht per MQTT angebunden werden soll. Stattdessen diese Zeile ergaenzen:

```
api:
```

Hiermit wird die native API geladen, welche fuer die direkte Kommunikation mit Home Assistant zum Einsatz kommt.

3. Im YAML gibt es ein paar Platzhalter vom Typ `!secret <KEY>`. Für gewöhnlich legt man ins gleiche Verzeichnis, wo man auch seine YAML-Datei aufbewahrt eine Datei mit dem Namen `secrets.yaml` ab. Der Inhalt ist eine Liste aus Schlüssel+Wert:

```
wifi_ssid: DEIN_WLAN_NETZ
wifi_password: DEIN_WLAN_PASSWORT
dl24_mac_address: DIE_MAC_ADDRESSE_DEINES_DL24
``` 

Sollte man im Home Assistant Dashboard diese Datei nicht anlegen/ablegen können, kannst du auch die Secret-Instruktionen löschen und direkt mit Werten bestücken (in der YAML):

```
wifi:
  ssid: "DEIN_WLAN_NETZ"
  password: "DEIN_WLAN_PASSWORT"
``` 

```
substitutions:
  name: atorch-dc-meter
  external_components_source: github://syssi/esphome-atorch-dl24@main
  dl24_mac_address: "ff:ee:dd:cc:bb:aa"
```

4. Nun speicherst du die YAML-Datei / Konfiguration und bittest ESPHome sie zu compilieren und auf einen ESP32 zu spielen.
5. Sobald sich die neue ESPHome-Node im Netzwerk eingebucht hat sollte Home Assistant sie bereits links unten als neues verfügbares ESPHome-Gerät ankündigen. Sollte die Benachrichtigung nicht erscheinen, kannst du die Node manuell über 

```
-> Einstellungen
-> Geräte
-> Intergration hinzufügen
-> ESPHome
-> Weiteres Gerät einrichten
-> Host: "atorch-dc-meter.local"
-> Absenden
``` 

in einer Home Assistant Instanz bekannt machen. Anschließend findest du die Node unter "Geräte" samt aller Sensorwerte und Bedienelementen.
